analysis_driver/dataset.py

```diff
@@ -657,6 +669,10 @@ def initialise_entity(self):
             'dataset_type': self.dataset.type,
             'dataset_name': self.dataset.name
         }
+
+        if self.dataset.type in ['project', 'sample']:
+            entity.update({'data_source': self.dataset.data_source})
```

Comment:
This is going to raise NotImplementedError in RunDataset's case

Reply:
This I think would not raise NotImplementedError in RunDataset due to the if statement - unless I'm missing something?
I have added data_source to RunDataset but returning None instead to not have to check for dataset type

Reply:
that is indeed correct